packages/internal/src/generate/typeDefinitions.ts

```diff
@@ -262,29 +266,88 @@ const runCodegenGraphQL = async (
       namingConvention: 'keep', // to allow camelCased query names
       scalars: {
         // We need these, otherwise these scalars are mapped to any
-        // @TODO is there a way we can use scalars defined in packages/graphql-server/src/rootSchema.ts
+        // @TODO is there a way we can use scalars defined in
+        // packages/graphql-server/src/rootSchema.ts
         BigInt: 'number',
         DateTime: 'string',
         Date: 'string',
         JSON: 'Record<string, unknown>',
         JSONObject: 'Record<string, unknown>',
         Time: 'string',
       },
-      omitOperationSuffix: true, // prevent type names being PetQueryQuery, RW generators already append Query/Mutation/etc
+      // prevent type names being PetQueryQuery, RW generators already append
+      // Query/Mutation/etc
+      omitOperationSuffix: true,
     },
     generates,
     silent: false,
     errorsOnly: true,
   }
 
+  return codegenConfig
+}
+
+export const generateTypeDefGraphQLWeb = async () => {
+  const rwjsPaths = getPaths()
+  const documentsGlob = './web/src/**/!(*.d).{ts,tsx,js,jsx}'
+
+  const generates = {
+    [path.join(rwjsPaths.web.types, 'graphql.d.ts')]: {
+      documents: documentsGlob,
+      plugins: [
+        {
+          typescript: {
+            enumsAsTypes: true,
+          },
+        },
+        'typescript-operations',
+      ],
+    },
+  }
+
+  const options = generateLoadTypedefsConfig(generates)
+
+  try {
+    await loadDocuments([documentsGlob], options)
+  } catch {
+    // No GraphQL documents present (web), no need to try to run codegen
+    return []
```

Comment:
This seems a little problematic... no? What happens if you have different errors? Like incorrect directives, or scalars that haven't been defined correctly?

Reply:
Remember that `loadDocuments` only loads the documents, it doesn't validate the documents.

I added three new tests for different errors to make sure we're not catching them here.